packages/block-library/src/navigation/edit/placeholder/index.js

```diff
@@ -50,30 +66,194 @@ export default function Placeholder( {
+		const navigationMenu = await createNavigationMenu(
+			navigationMenuTitle,
+			blocks
+		);
+		onFinish( navigationMenu );
+	};
+
+	const {
+		isResolvingPages,
+		menus,
+		isResolvingMenus,
+		menuItems,
+		hasResolvedMenuItems,
+		hasPages,
+		hasMenus,
+	} = useNavigationEntities( selectedMenu );
+
+	const isStillLoading = isResolvingPages || isResolvingMenus;
+
+	const createFromMenu = useCallback(
+		( name ) => {
+			const { innerBlocks: blocks } = menuItemsToBlocks( menuItems );
+			onFinishMenuCreation( name, blocks );
+		},
+		[ menuItems, menuItemsToBlocks, onFinish ]
+	);
+
+	const onCreateFromMenu = ( name ) => {
+		// If we have menu items, create the block right away.
+		if ( hasResolvedMenuItems ) {
+			createFromMenu( name );
+			return;
+		}
+
+		// Otherwise, create the block when resolution finishes.
+		setIsCreatingFromMenu( true );
+		// Store the name to use later.
+		setMenuName( name );
+	};
+
+	const onCreateEmptyMenu = ( name ) => {
+		onFinishMenuCreation( name, [] );
+	};
+
+	const onCreateAllPages = ( name ) => {
+		const block = [ createBlock( 'core/page-list' ) ];
+		onFinishMenuCreation( name, block );
+		setIsNewMenuModalVisible( true );
+	};
+
+	useEffect( () => {
+		// If the user selected a menu but we had to wait for menu items to
+		// finish resolving, then create the block once resolution finishes.
+		if ( isCreatingFromMenu && hasResolvedMenuItems ) {
+			createFromMenu( menuName );
+			setIsCreatingFromMenu( false );
+		}
+	}, [ isCreatingFromMenu, hasResolvedMenuItems, menuName ] );
+
+	const toggleProps = {
+		variant: 'primary',
+		className: 'wp-block-navigation-placeholder__actions__dropdown',
+	};
+
+	const { navigationMenus } = useNavigationMenu();
+
 	return (
 		<>
-			{ step === PLACEHOLDER_STEPS.selectNavigationPost && (
-				<SelectNavigationMenuStep
-					onCreateNew={ ( newTitle ) => {
-						setNavigationMenuTitle( newTitle );
-						setStep( PLACEHOLDER_STEPS.createInnerBlocks );
-					} }
-					onSelectExisting={ ( navigationMenu ) => {
-						onFinish( navigationMenu );
-					} }
-					canSwitchNavigationMenu={ canSwitchNavigationMenu }
-					hasResolvedNavigationMenu={ hasResolvedNavigationMenu }
-				/>
+			{ ! hasResolvedNavigationMenu && <PlaceholderPreview isLoading /> }
+			{ hasResolvedNavigationMenu && (
+				<Placeholder className="wp-block-navigation-placeholder">
+					<PlaceholderPreview />
+
+					<div className="wp-block-navigation-placeholder__controls">
+						{ isStillLoading && (
+							<div>
+								<Spinner />
+							</div>
+						) }
+						{ ! isStillLoading && (
+							<div className="wp-block-navigation-placeholder__actions">
+								<div className="wp-block-navigation-placeholder__actions__indicator">
+									<Icon icon={ navigation } />{ ' ' }
+									{ __( 'Navigation' ) }
+								</div>
+								{ hasMenus || canSwitchNavigationMenu ? (
+									<DropdownMenu
+										text={ __( 'Add existing menu' ) }
```

Comment:
I'm not sure 'Add' is the right word now that the `wp_navigation` menus are in this dropdown, since the user is directly editing the menu. Maybe 'Select existing menu'.